Bugzilla/App.pm

```diff
@@ -205,17 +196,17 @@ sub startup {
 
 sub setup_routes {
   my ($self) = @_;
-
   my $r = $self->routes;
-  Bugzilla::App::API->setup_routes($r);
-  Bugzilla::App::BouncedEmails->setup_routes($r);
-  Bugzilla::App::CGI->setup_routes($r);
-  Bugzilla::App::Main->setup_routes($r);
-  Bugzilla::App::OAuth2::Provider::Clients->setup_routes($r);
-  Bugzilla::App::SES->setup_routes($r);
-  Bugzilla::App::BMO::AntiSpam->setup_routes($r);
-  Bugzilla::App::BMO::NewRelease->setup_routes($r);
-  Bugzilla::App::MFA::Duo->setup_routes($r);
+
+  # Load controller modules and their routes
+  foreach my $module (find_modules('Bugzilla::App::Controller', {recursive => 1}))
+  {
+    require_module($module);
+    my $controller = $module->new;
+    if ($controller->can('setup_routes')) {
+      $controller->setup_routes($r);
+    }
```

Comment:
Should the absence of a `setup_routes` function be an error? Or maybe we should log it for debugging purposes etc?